internal/client/client_batch.go

```diff
@@ -321,16 +376,101 @@ func (a *batchConn) fetchMorePendingRequests(
 			}
 			a.reqBuilder.push(entry)
 		default:
-			return
+			if yield {
+				return
+			}
+			// yield once to batch more requests.
+			runtime.Gosched()
+			yield = true
 		}
 	}
 }
 
 const idleTimeout = 3 * time.Minute
 
+var (
+	presetBatchPolicies = map[string]turboBatchOptions{
+		config.BatchPolicyBasic:    {},
+		config.BatchPolicyStandard: {V: turboBatchTimeBased, T: 0.0001, N: 5, W: 0.2, P: 0.8, Q: 0.8},
+		config.BatchPolicyPositive: {V: turboBatchAlways, T: 0.0001},
+	}
+)
+
+const (
+	turboBatchAlways = iota
+	turboBatchTimeBased
+	turboBatchProbBased
+)
+
+type turboBatchOptions struct {
+	V int     `json:"v"`
+	N int     `json:"n,omitempty"`
+	T float64 `json:"t,omitempty"`
+	W float64 `json:"w,omitempty"`
+	P float64 `json:"p,omitempty"`
+	Q float64 `json:"q,omitempty"`
+}
+
+type turboBatchTrigger struct {
```

Comment:
Need to add comments from the design doc here to introduce the `turboBatchTrigger` and `turboBatchOptions`  concepts briefly.

The detailed design document is also neede to be referenced in the github issue.